src/core/ddsi/src/ddsi_cdrstream.c

```diff
@@ -354,6 +354,96 @@ static uint32_t get_type_size (enum dds_stream_typecode type)
   return (uint32_t)1 << ((uint32_t) type - 1);
 }
 
+static uint32_t get_elem_size (uint32_t insn, const uint32_t * __restrict ops)
+{
+  uint32_t elem_sz;
+  switch (DDS_OP_SUBTYPE (insn))
+  {
+    case DDS_OP_VAL_1BY: case DDS_OP_VAL_2BY: case DDS_OP_VAL_4BY: case DDS_OP_VAL_8BY: case DDS_OP_VAL_ENU:
+      elem_sz = get_type_size (DDS_OP_SUBTYPE (insn));
+      break;
+    case DDS_OP_VAL_STR:
+      elem_sz = sizeof (char *);
+      break;
+    case DDS_OP_VAL_BST: case DDS_OP_VAL_SEQ: case DDS_OP_VAL_ARR: case DDS_OP_VAL_UNI: case DDS_OP_VAL_STU:
+      if (DDS_OP_TYPE (insn) == DDS_OP_VAL_ARR)
+        elem_sz = ops[4];
+      else
+        abort ();
+      break;
+    case DDS_OP_VAL_EXT:
+    default:
```

Comment:
I'd prefer to get rid of the `default` here (and in similar cases): then the compiler will warn if some cases are skipped.

Reply:
Good point, in this case there shouldn't be a `default`, fixed in 9c53626. In other cases, e.g. where only the `1BY` ... `ENU` values are included, it's better to leave out the other (7) values I think?